### PR TITLE
Account update delete

### PR DESCRIPTION
**Pull request by FedeOt**
-Adds a component to update and delete the accounts
-Adds a new feature in the login to know who's logged
-Adds unit test to the update component.

### CODE REVIEW

File: src/axios/Instance.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import axios from 'axios'
 
 
 export const getAuthHeader = () => `Bearer ${sessionStorage.getItem('token')}`
+export const getRole = () => sessionStorage.getItem('role');
```
Comment (ghost) [inline src/axios/Instance.js:5]: I'd put it in a login-related functions file.

File: src/api/index.js [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+      return axiosClient.get('user/account/checking', {
+        headers: { Authorization: getAuthHeader() },
+      });
+    }
+
+    if(role === 'ROLE_ADMIN'){
+      return axiosClient.get('/account/checking', {
+        headers: { Authorization: getAuthHeader() },
+      });
+    }
```
Comment (ghost) [inline src/api/index.js]: You can use the _Ternary Opeator_ here.

File: src/api/index.js [hunk 3]
```diff
@@ -14,4 +26,14 @@ export const createCheckingAccount = (account) =>
...
+
+export const deleteAccount = (id) =>
+    axiosClient.delete(`/account/${id}`,{
+      headers: {Authorization: getAuthHeader()}
+    })
+
+export const updateAccount = (id,data) =>
+    axiosClient.put(`/account/${id}`,{},
+    {headers:{Authorization: getAuthHeader()},
+    params:{newName:data}});
```
Comment (ghost) [inline src/api/index.js]: If `data` only stores the Account Name, I'd name this parameter `name` instead.

File: src/index.css [hunk 4]
```diff
@@ -73,6 +73,8 @@ code {
   padding: 10px;
 }
 
+
+
```
Comment (ghost) [inline src/index.css]: Could you remove these spaces? :)

File: src/components/UpdateInput.jsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    setInputValue(e.target.value)
+    
+  }
+
+  const handleSubmit = async(e) =>{
+      e.preventDefault();
+      if(inputValue.trim() === ''){
+        return setNewNameEmpty(true); 
+      }
+      console.log(inputValue);
```
Comment (ghost) [inline src/components/UpdateInput.jsx]: This line isn't necessary anymore, right?

File: src/components/UpdateInput.jsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    setNewNameEmpty(false); 
+    setInputValue(e.target.value)
+    
+  }
+
+  const handleSubmit = async(e) =>{
+      e.preventDefault();
+      if(inputValue.trim() === ''){
+        return setNewNameEmpty(true); 
+      }
```
Comment (ghost) [inline src/components/UpdateInput.jsx:21]: You can use the _Ternary Operator_ here.

File: src/components/UpdateInput.jsx [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+          {
+            newNameEmpty && <span className='text-danger ms-4'>New name is required</span>
+          }
+      </div>
+        
+      <hr />
+  
+    </div>
+  )
+}
```
Comment (ghost) [inline src/components/UpdateInput.jsx:52]: Format this file, please.

File: src/components/AccountCard.jsx [hunk 8]
```diff
@@ -1,18 +1,54 @@
...
+import { useState } from 'react';
+import { deleteAccount } from '../api'
+import { getRole } from '../axios/Instance';
 import { formatDate } from '../moment/formatDate'
+import { UpdateInput } from './UpdateInput';
+
+
+export const AccountCard = (props) => {
+
+  const [inputOpen,setInputOpen] = useState(false);
```
Comment (ghost) [inline src/components/AccountCard.jsx:10]: Is this the New Name input?

File: src/components/AccountCard.jsx [hunk 9]
```diff
@@ -1,18 +1,54 @@
...
+
+
+export const AccountCard = (props) => {
+
+  const [inputOpen,setInputOpen] = useState(false); 
+
+  const handleDelete = async(id) =>{
+
+    const response = await deleteAccount(id);
+    console.log(response.status);
```
Comment (ghost) [inline src/components/AccountCard.jsx]: Can you remove this line?

File: src/tests/AccountUpdate.test.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import '@testing-library/jest-dom/extend-expect'
+import { fireEvent, render, waitFor } from '@testing-library/react'
+import { getRole } from '../axios/Instance';
+import { AccountCard } from '../components/AccountCard';
+
+jest.mock('../axios/instance.js');
+
+const INPUTVALUE = "New account";
```
Comment (ghost) [inline src/tests/AccountUpdate.test.js]: Use `camelCase`.

File: src/tests/AccountUpdate.test.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+    const updateButton = updateComponent.getByTestId('update-button');  
+    fireEvent.click(updateButton); 
+
+    const updateInput = updateComponent.getByTestId('update-input');
+    const submitButton = updateComponent.getByTestId('submit'); 
+    fireEvent.change(updateInput,{target:{value:INPUTVALUE}}); 
+    fireEvent.click(submitButton);
+
+    await waitFor(()=>{
+        expect(updateInput).not.toBeInTheDocument();
```
Comment (ghost) [inline src/tests/AccountUpdate.test.js]: You should add an assertion to check if the Account Name changed after updating it.

File: src/tests/AccountUpdate.test.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    fireEvent.click(submitButton);
+
+    await waitFor(()=>{
+        expect(updateInput).not.toBeInTheDocument(); 
+    })
+
+
+  
+
+})
```
Comment (ghost) [inline src/tests/AccountUpdate.test.js]: Remember to format this file, please! :)

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/tests/AccountUpdate.test.js (hunk 12 above)
Comment (FedeOt) [reply src/tests/AccountUpdate.test.js]: Done!

File: src/tests/AccountUpdate.test.js (hunk 10 above)
Comment (FedeOt) [reply src/tests/AccountUpdate.test.js]: Done!

File: src/components/AccountCard.jsx (hunk 9 above)
Comment (FedeOt) [reply src/components/AccountCard.jsx]: Do you mean the console.log?

File: src/components/AccountCard.jsx (hunk 8 above)
Comment (FedeOt) [reply src/components/AccountCard.jsx:10]: That useState is opening and closing the new name input, depending on the state

File: src/components/UpdateInput.jsx (hunk 5 above)
Comment (FedeOt) [reply src/components/UpdateInput.jsx]: the console.log is not necessary anymore

File: src/api/index.js (hunk 3 above)
Comment (FedeOt) [reply src/api/index.js]: Done!

File: src/index.css (hunk 4 above)
Comment (FedeOt) [reply src/index.css]: Yes

File: src/components/AccountCard.jsx (hunk 9 above)
Comment (ghost) [reply src/components/AccountCard.jsx]: Yes!

**Review by ghost — APPROVED** [review]
Looks good to me!

**Pull request merged**